File: ruby/bob/bob.rb
class Bob
  def hey(remark)
    if remark.strip.empty?
      'Fine. Be that way!'
    elsif remark.upcase == remark && remark.downcase != remark
      'Whoa, chill out!'
    elsif remark.end_with?('?')
      'Sure.'
    else
      'Whatever.'
    end
  end
end

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (!/\S/.test(input)) {
    return 'Fine. Be that way!';
  } else if (/[a-zA-Z]/.test(input) && input.toUpperCase() === input) {
    return 'Whoa, chill out!';
  } else if (input.substr(-1) === '?') {
    return 'Sure.';
  } else {
    return 'Whatever.';
  }
};

module.exports = Bob;
